WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -452,6 +493,26 @@ private extension ProductFormViewModel {
         let oldProductWithNewName = EditableProductModel(product: oldProduct.product.copy(name: newProduct.name))
         return oldProductWithNewName == newProduct && newProduct.name != oldProduct.name
     }
+
+    /// Updates the `newVariationsPriceSubject`and `actionsFactory` to the latest variations price information.
+    /// Returns weather the variation
+    ///
+    func updateVariationsPriceState() {
+        updateActionsFactory()
+        newVariationsPriceSubject.send(())
+    }
+
+    /// Calculates the variations price state for the current fetched variations.
+    ///
+    func calculateVariationPriceState() -> ProductFormActionsFactory.VariationsPrice {
+        // If there are no fetched variations we can't be sure of it's price state
+        guard variationsResultsController.fetchedObjects.isNotEmpty else {
+            return .unknown
+        }
+
+        let someMissingPrice = variationsResultsController.fetchedObjects.contains { $0.regularPrice.isNilOrEmpty }
```

Comment:
This is just a thought exercise.

Not sure if this is going to work but if it will, it will probably perform better. We _could_ include this `contains` predicate in the `ResultsController` predicate instead. 

```suggestion
        let predicate = NSPredicate(format: "product.siteID = %ld AND product.productID = %ld AND (product.regularPrice == nil OR product.regularPrice == '')", product.siteID, product.productID)
```

This _should_ return only the list of variations that don't have prices. And then we'd be able to use `isEmpty` and no mapping will ever happen. 

The problem with this is there's no way to check the other condition above, if the product has variations or not. So it's not enough. 

Something to think about. Our `ResultsController` API is unwieldy and could be better. Sometimes I wish I can just run SQL. 🤪